### PR TITLE
Windows: `CommandExt::async_pipes`

### PR DESCRIPTION
**Pull request by ChrisDenton**
Discussed in https://github.com/tokio-rs/tokio/issues/4670 was the need for third party crates to be able to force `process::Command::spawn` to create pipes as async.

This implements the suggestion for a `async_pipes` method that gives third party crates that option.

# Example:

```rust
use std::process::{Command, Stdio};

Command::new("cmd")
    .async_pipes(true)
    .stdin(Stdio::piped())
    .stdout(Stdio::piped())
    .stderr(Stdio::piped())
    .spawn()
    .unwrap();
```

### CODE REVIEW

**Comment by ChrisDenton** [comment]
r? rust-lang/libs-api

**Comment by dtolnay** [comment]
Seems okay to me. @m-ou-se could you take a look since you requested this API in https://github.com/tokio-rs/tokio/issues/4670#issuecomment-1129666326?

r? m-ou-se

File: library/std/src/os/windows/process.rs [hunk 1]
```diff
@@ -161,6 +161,37 @@ pub trait CommandExt: Sealed {
...
+    ///
+    /// By default [`process::Command`] may choose to use pipes where both ends
+    /// are opened for synchronous read or write operations. By using
+    /// `async_pipes(true)`, this behavior is overridden so that our side is
+    /// always async.
+    ///
+    /// This is important because if doing async I/O a pipe or a file has to be
+    /// opened for async access.
+    ///
+    /// The end of the pipe set to the child process will always be synchronous
```
Comment (m-ou-se) [inline library/std/src/os/windows/process.rs:175]:
```suggestion
    /// The end of the pipe sent to the child process will always be synchronous
```

**Review by m-ou-se — APPROVED** [review]
Looks good to me. r=me after minor changes.

File: library/std/src/os/windows/process.rs [hunk 2]
```diff
@@ -161,6 +161,37 @@ pub trait CommandExt: Sealed {
...
+    /// regardless of this option.
+    ///
+    /// # Example
+    ///
+    /// ```
```
[the remark this answers is not in the document]
Comment (m-ou-se) [reply library/std/src/os/windows/process.rs:180]: Oh, right. Never mind. :)

**Comment by ChrisDenton** [comment]
@bors r-

Oops, forgot to post the tracking issue for this one. I really shouldn't try to do two things at once. 🙂

**Pull request merged**